spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/ConfigUtils.java

```diff
@@ -130,4 +149,43 @@ public static String sourceName(String target, String applicationName, String na
 		return target + PROPERTY_SOURCE_NAME_SEPARATOR + applicationName + PROPERTY_SOURCE_NAME_SEPARATOR + namespace;
 	}
 
+	public static final class Prefix {
+
+		/**
+		 * prefix has not been provided.
+		 */
+		public static final Prefix UNSET = new Prefix(() -> {
```

Comment:
its not only that we are not providing one, its that we don't want one either right?

Would it make sense to just use `null`?  It almost seems like it being "unset" implies something is wrong, when in fact that is ok.

Or I guess another option we to set it to the empty string

Reply:
> its not only that we are not providing one, its that we don't want one either right?

entirely correct. 

I _think_ this is how you see it, right?

```
/**
 * prefix has not been provided.
 */
public static final Prefix UNSET = new Prefix(() -> "");
```

If so, this is a very valid point, agreed.

Reply:
Yes but maybe its best to name it `DEFAULT`?